components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -2776,6 +2784,109 @@ impl ToAnimatedZero for IntermediateSVGPaintKind {
     }
 }
 
+impl Into<NumberOrPercentage> for NonNegativeLengthOrPercentage {
+    fn into(self) -> NumberOrPercentage {
+        self.0.into()
+    }
+}
+
+impl Into<NumberOrPercentage> for NonNegativeNumber {
+    fn into(self) -> NumberOrPercentage {
+        self.0.into()
+    }
+}
+
+impl Into<NumberOrPercentage> for LengthOrPercentage {
+    fn into(self) -> NumberOrPercentage {
+        match self {
+            LengthOrPercentage::Length(len) => NumberOrPercentage::Number(len.to_f32_px()),
+            LengthOrPercentage::Percentage(p) => NumberOrPercentage::Percentage(p),
+            LengthOrPercentage::Calc(_) => {
+                // TODO: We need to treat calc value.
+                // https://bugzilla.mozilla.org/show_bug.cgi?id=1386967
```

Comment:
Why do we not expect `calc()` values here? Why should they be rejected? Why does the type have a `Calc` variant if it should not? Seems like the SVG types are badly designed. :(